### PR TITLE
Split release notes in different files (rc)

### PR DESCRIPTION
**Pull request by rmoyard**
Description of the Change:

Split the changelog into separate file for each release, easier to read and help for the rc sync action.

### CODE REVIEW

File: doc/dev/release_notes.rst [hunk 1]
```diff
@@ -3,4 +3,30 @@ Release notes
 
 This page contains the release notes for Catalyst.
 
-.. mdinclude:: ../changelog.md
+.. mdinclude:: ../releases/changelog-0.8.0.md
```
Comment (josh146) [inline doc/dev/release_notes.rst:6]: Need to remember to also show `-dev.md` here when we merge this to `main`!

**Review by josh146 — APPROVED** [review]
Thanks Romain, I only did a quick skim through https://xanaduai-pennylane--1070.com.readthedocs.build/projects/catalyst/en/1070/dev/release_notes.html but it seems to look okay

**Comment by dime10** [comment]
Do we need this in the rc branch? It doesn't seem release critical so I would favour merging to main only.

We'll also have to update some docs in a few places to go along with this, @rmoyard can you check if there is any mention of the changelog in the repo (like the pr template, changelog reminder action, etc), and probably the release guide in the other repo as well.

**Comment by rmoyard** [comment]
Yes I think it is needed because I want to merge the rc sync, otherwise there will bug clashes.

I am updating the release guide after we merge the rc sync.

The changelog remainder needs to be updated.

**Comment by rmoyard** [comment]
[sc-72457]

File: doc/dev/release_notes.rst (hunk 1 above)
Comment (rauletorresc) [reply doc/dev/release_notes.rst:6]: I think this is the only point to resolve in order to merge this PR with RC @josh146 do you approve?

**Review by rauletorresc — APPROVED** [review]

**Pull request merged**